src/pydata_sphinx_theme/assets/styles/components/_search.scss

```diff
@@ -39,46 +56,36 @@
     &::-webkit-search-decoration {
       appearance: none;
     }
+
+    &:focus,
+    &:focus-visible {
+      color: var(--pst-color-text-muted);
+    }
   }
 
   // Shows off the keyboard shortcuts for the button
   .search-button__kbd-shortcut {
     display: flex;
-    position: absolute;
```

Comment:
I think `position: absolute` was actually a bad idea because it allows a the search terms in the input box (think narrow screen rather than super long search string) to bleed into the span displaying the keyboard shortcut.